src/patternfly/components/NotificationBadge/notification-badge.scss

```diff
@@ -21,6 +21,7 @@
   --pf-c-notification-badge--m-read--after--BackgroundColor: transparent;
   --pf-c-notification-badge--m-read--after--BorderColor: transparent;
   --pf-c-notification-badge--m-unread--after--BackgroundColor: var(--pf-global--active-color--400);
+  --pf-c-notification-badge--m-unread--c-button--m-masthead--after--BorderColor: var(--pf-global--BackgroundColor--dark-200);
```

Comment:
Any suggestions on naming this variable. Since the button wraps around the notification badge component, I'm wondering if it should be `--pf-c-button--m-masthead__notification-badge--m-unread--after--BorderColor` because then it matches the selector order. But it seems odd to have a variable in a component start with another variable's name. I think this is the first time this has occurred. Maybe theres another solution here. @mcoker @mattnolting @srambach

Reply:
You could move these styles to the button component since the styles you're trying to match come from the button.

But if I'm not mistaken, like we talked about in the last css meeting, I think you're creating a variation of the button here instead of just overwriting the button's styles in the page component? But then you're overwriting the notification badge styles based on being inside of another component (or modifying the notification badge styles in the button component, if you moved the styles). You could create a modifier for the notification badge for this use case. Though I'm wondering if it might be better just to overwrite the button and notification badge styles in the masthead instead. Maybe it wasn't a great idea to create a modifier for use when we need to style one component in another, unless it's a repeatable variation we'll use in other components.

Reply:
This background style is for all icons in the masthead so I can't create a modifier for just the notification badge, I don't think, that's why I targeted the plain modification of the button. 

@mcarrano  @maryshak1996 can you for see this background style for icons outside of the masthead?  Or is it only for the icons in the masthead?

Reply:
Couldn't you use `.pf-m-masthead` with the notification drawer as you've done for the button? I'm also not sure about masthead as the class. What if we have 2 modifiers for buttons in the masthead? That class also doesn't age well if we end up doing this somewhere else.

Reply:
Okay so are you suggesting to just overwrite the styles of buttons in the masthead of the page component, and then if another use case comes up for them, we can look at making it its own example in the button component ?

Reply:
I think so, it seems simpler and more appropriate just to make the change in the context of where it needs to change than trying to abstract it out this way. What do you think?

Reply:
Yeah I agree!